navapbc/labs-decision-support-tool
# feat: Add LLM selection in Chainlit

## Ticket

https://navalabs.atlassian.net/browse/DST-270

## Changes

- Allow default LLM to be set via env variable.
- Allow the user to select their preferred LLM, by showing a dropdown with the formatted names from `get_models()`.
- When submitting a generation request to the LLM, use the selected LLM model.

## Context for reviewers

Prior to this, the LLM model was hardcoded.

## Testing

1. Start Ollama
2. Add `OLLAMA_HOST=http://host.docker.internal:11434` to `.env` so that there are multiple LLMs to choose from. 
3. Start Chainlit UI, click on the Chainlit settings, select an LLM, submit a request and confirm that the LLM response is generated from the chosen LLM.
   - examine logs to see that the chosen LLM was used, such as `Calling ollama/llama2:latest for query: ...`
<img width="445" alt="image" src="https://github.com/user-attachments/assets/157937b7-45ae-4613-ad10-839ad85b5259">

To set the default LLM, `export LLM='ollama/llama2:latest'` then restart Chainlit.
